tchannel-core/src/main/java/com/uber/tchannel/api/SubChannel.java

```diff
@@ -64,7 +64,7 @@
     private final @NotNull TChannel topChannel;
     private final @NotNull PeerManager peerManager;
     private final long initTimeout;
-    private final Connection.Direction preferredDirection;
+    private final @NotNull Connection.Direction preferredDirection;
```

Comment:
i left this one unannotated because strictly speaking there is no requirement for it to be non-null.
so, technically, there might be some (currently working) client code that's passing a `null` here, and this change can break it.